x/crosschain/keeper/abci.go

```diff
@@ -175,3 +200,18 @@ func CheckAndUpdateCctxGasPrice(
 
 	return gasPriceIncrease, additionalFees, nil
 }
+
+// CheckAndUpdateCCTXGasPriceBTC updates the fee rate for the given Bitcoin chain CCTX
+func CheckAndUpdateCCTXGasPriceBTC(
+	ctx sdk.Context,
+	k Keeper,
+	medianGasPrice math.Uint,
+	cctx types.CrossChainTx,
+) (math.Uint, math.Uint, error) {
+	// zetacore simply update 'GasPriorityFee', and zetaclient will use it to schedule RBF tx
+	// there is no priority fee in Bitcoin, the 'GasPriorityFee' is repurposed to store latest fee rate in sat/vB
+	cctx.GetCurrentOutboundParam().GasPriorityFee = medianGasPrice.String()
```

Comment:
Where does the additional fees are taken from? If it's fully inferred by the protocol there might be a insolvency issue
For EVM with use gas stability pool. For Bitcoin we can have a gas stability pool as well, it would be not autoamatically filled like Ethereum, but could still be filled manually by sending BTC to the pool